source/Calamari.Shared/Integration/Certificates/WindowsX509CertificateStore.cs

```diff
@@ -516,11 +516,15 @@ static CryptoKeySecurity GetCspPrivateKeySecurity(SafeCertContextHandle certific
             }
         }
 
-        static IList<Org.BouncyCastle.X509.X509Certificate> GetCertificatesToImport(byte[] pfxBytes, string password)
+        static IList<Org.BouncyCastle.X509.X509Certificate> GetCertificatesToImport(byte[] pfxBytes, string? password)
         {
             using (var memoryStream = new MemoryStream(pfxBytes))
             {
-                var pkcs12Store = new Pkcs12Store(memoryStream, password?.ToCharArray() ?? "".ToCharArray());
+                // The latest version of BouncyCastle fails if the cert doesn't require a key, but we pass an empty array key.
+                // Will issue a PR to make this configurable at least in a way that doesn't require writing environment variables.
+                Environment.SetEnvironmentVariable("Org.BouncyCastle.Pkcs12.IgnoreUselessPassword", "true");
+                var pkcs12Store = new Pkcs12StoreBuilder().Build();
+                pkcs12Store.Load(memoryStream, password?.ToCharArray() ?? "".ToCharArray());
```

Comment:
Not entirely sure why they [used an environment variable](https://github.com/bcgit/bc-csharp/blob/master/crypto/src/pkcs/Pkcs12Store.cs#L219) rather than something with more control like an instance or static property.

In the older version of BouncyCastle this password check did not take place.